packages/dvilela/connections/twikit/connection.py

```diff
@@ -371,7 +371,9 @@ async def post_tweet(self, **kwargs: Any) -> Optional[str]:
                 self.logger.info(f"Posting: {kwargs}")
                 result = await self.client.create_tweet(**kwargs)
                 tweet_id = result.id
-                break
+                if tweet_id is not None:
+                    self.logger.info(f"Tweet created with tweet ID: {tweet_id}")
+                    break
```

Comment:
We need to increase retries after the if block, no? If not, we might get stuck here forever. I think we need to increase retries inside a finally clause. Then, after the while, check whether the tweet_id is None and return an error if so